lib/wordpress.ts

```diff
@@ -30,169 +31,105 @@ export async function getAllPosts(filterParams?: {
   tag?: string;
   category?: string;
 }): Promise<Post[]> {  
-  const url = getUrl("/wp-json/wp/v2/posts", { author: filterParams?.author, tags: filterParams?.tag, categories: filterParams?.category });
-  const response = await fetch(url);
-  const posts: Post[] = await response.json();
-  return posts;
+  return wpFetch<Post[]>("/wp-json/wp/v2/posts", { author: filterParams?.author, tags: filterParams?.tag, categories: filterParams?.category });
```

Comment:
I'm not very familiar with WordPress, but if the `/wp-json/wp/v2/` part of the path is constant, that could also be moved into the `wpFetch` wrapper. Repeating it doesn't really seem to be serving any purpose.